src/util.c

```diff
@@ -17,3 +17,36 @@ long parse_int(char *str) {
 
     return parsed;
 }
+
+unsigned long parse_buttons_numbers(char *str,
+ignore_buttons_t *ignore_buttons) {
+    ignore_buttons->buttons = (unsigned int *)calloc(10, sizeof(unsigned int));
+    // no check for "more then UCHAR_MAX"
```

Comment:
(nit: then -> than)

But actually, I think we should just remove this "batching" logic for the allocated size and reallocate as we go along. We usually only expect a few entries here, we *never* expect many, so having to do 2–3 reallocs is totally fine since it only happens on startup anyway and reduces the complexity of this function.